net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc

```diff
@@ -4409,4 +4415,834 @@ EOD;
+		}
+	}
+
+	// EAP Maximum Sessions Tracking Per Server
+	if ($post['vareapconfmaxsessions'] != '' && !is_numericint($post['vareapconfmaxsessions'])) {
+		$input_errors[] = "The 'Maximum Sessions Tracking Per Server' field must contain an integer value.";
+	}
+
+	// EAP Certificates for TLS sanity checking
+	if ($post['vareapconfchoosecertmanager'] == 'on') {
+		// EAP Private Key Password should be empty if using pfSense Cert Manager
+		if ($post['vareapconfprivatekeypassword'] != '') {
+			$input_errors[] = "When using pfSense Certificate Management, the 'Private Key Password' field must be empty.";
+		}
+	} else {
+		// SSL CA Certificate, SSL Revocation List and SSL Server Certificate should be 'none' if using the deprecated FreeRADIUS Cert Manager
+		if ($post['ssl_ca_cert'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL CA Certificate' field must be set to 'none'.";
+		}
+		if ($post['ssl_ca_crl'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL Revocation List' field must be set to 'none'.";
+		}
+		if ($post['ssl_server_cert'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL Server Certificate' field must be set to 'none'.";
+		}
+	}
+
+	// EAP-TLS Fragment Size
+	if ($post['vareapconffragmentsize'] != '') {
+		if (!is_numericint($post['vareapconffragmentsize'])) {
+			$input_errors[] = "The 'Fragment Size' field must contain an integer value.";
+		} elseif ($post['vareapconffragmentsize'] > 4096) {
+			$input_errors[] = "The 'Fragment Size' field must contain an integer value not higher than 4096.";
+		}
+	}
+
+	// EAP-TLS Cache Lifetime
+	if ($post['vareapconfcachelifetime'] != '' && !is_numericint($post['vareapconfcachelifetime'])) {
+		$input_errors[] = "The EAP-TLS Cache 'Lifetime' field must contain an integer value.";
+	}
+
+	// EAP-TLS Cache Max Entries
+	if ($post['vareapconfcachemaxentries'] != '' && !is_numericint($post['vareapconfcachemaxentries'])) {
+		$input_errors[] = "The EAP-TLS Cache 'Max Entries' field must contain an integer value.";
+	}
+
+	// EAP-TLS OCSP Responder
+	if (!empty($post['vareapconfocspurl'])) {
+		if (!filter_var($post['vareapconfocspurl'], FILTER_VALIDATE_URL)) {
+			$input_errors[] = "The 'OCSP Responder' field must contain a valid URL.";
+		}
+	}
+
+	/*
+	 * TODO: 
+	 * EAP-TLS - Country, State or Province, City, Organization, E-Mail Address, Common Name
+	 */
+
+}
+
+/* SQL settings input validation */
+function freeradius_validate_sql($post, &$input_errors) {
+
+	// SQL Server Address
+	if ($post['varsqlconfincludeenable'] == 'on') {
+		if (empty($post['varsqlconfserver'])) {
+			$input_errors[] = "The 'Server Address' field for SQL Server 1 must not be empty when 'SQL Support' for Server 1 is enabled.";
+		} elseif (!is_ipaddr($post['varsqlconfserver']) && !is_hostname($post['varsqlconfserver']) && !is_domain($post['varsqlconfserver'])) {
```

Comment:
is_hostname() also returns true for domain names (it uses is_domain() internally as part of its validation), so the is_domain() test isn't needed, just is_hostname() will catch short hostnames, FQDNs, and domain names.